templates/demo/app/dapp.js

```diff
@@ -1,161 +1,77 @@
 import EmbarkJS from 'Embark/EmbarkJS';
-import SimpleStorage from 'Embark/contracts/SimpleStorage';
+import Blockchain from './components/blockchain';
+import Whisper from './components/whisper';
+import Storage from './components/storage';
 
 import './dapp.css';
 
-var addToLog = function(id, txt) {
-  $(id + " .logs").append("<br>" + txt);
-};
+class App extends React.Component {
 
-// ===========================
-// Blockchain example
-// ===========================
-$(document).ready(function() {
+  constructor(props) {
+    super(props);
 
-  $("#blockchain button.set").click(function() {
-    var value = parseInt($("#blockchain input.text").val(), 10);
-
-    // If web3.js 1.0 is being used
-    if (EmbarkJS.isNewWeb3()) {
-      SimpleStorage.methods.set(value).send({from: web3.eth.defaultAccount});
-      addToLog("#blockchain", "SimpleStorage.methods.set(value).send({from: web3.eth.defaultAccount})");
-    } else {
-      SimpleStorage.set(value);
-      addToLog("#blockchain", "SimpleStorage.set(" + value + ")");
-    }
-
-  });
-
-  $("#blockchain button.get").click(function() {
-    // If web3.js 1.0 is being used
-    if (EmbarkJS.isNewWeb3()) {
-      SimpleStorage.methods.get().call(function(err, value) {
-        $("#blockchain .value").html(value);
-      });
-      addToLog("#blockchain", "SimpleStorage.methods.get(console.log)");
-    } else {
-      SimpleStorage.get().then(function(value) {
-        $("#blockchain .value").html(value.toNumber());
-      });
-      addToLog("#blockchain", "SimpleStorage.get()");
+    this.state = {
+      whisperEnabled: false,
+      storageEnabled: false
     }
-  });
-
-});
-
-// ===========================
-// Storage (IPFS) example
-// ===========================
-$(document).ready(function() {
-  // automatic set if config/storage.json has "enabled": true and "provider": "ipfs"
-  //EmbarkJS.Storage.setProvider('ipfs',{server: 'localhost', port: '5001'});
-
-  $("#storage .error").hide();
-  //EmbarkJS.Storage.ipfsConnection.version()
-  //  .then(function(){
-        $("#status-storage").addClass('status-online');
-        $("#storage-controls").show();
-   // })
-   // .catch(function(err) {
-   //   if(err){
-   //     console.log("IPFS Connection Error => " + err.message);
-   //     $("#storage .error").show();
-   //     $("#status-storage").addClass('status-offline');
-   //     $("#storage-controls").hide();
-   //   }
-   // });
+  }
 
-  $("#storage button.setIpfsText").click(function() {
-    var value = $("#storage input.ipfsText").val();
-    EmbarkJS.Storage.saveText(value).then(function(hash) {
-      $("span.textHash").html(hash);
-      $("input.textHash").val(hash);
-      addToLog("#storage", "EmbarkJS.Storage.saveText('" + value + "').then(function(hash) { })");
-    })
-    .catch(function(err) {
-      if(err){
-        console.log("IPFS saveText Error => " + err.message);
+  componentDidMount(){ 
+    let _this = this;
+
+    setTimeout(() => {
```

Comment:
Why do you need a timeout?

Reply:
`EmbarkJS.Messages.Providers.whisper` is not available when the component mounts